Src/FluentAssertions/Primitives/ObjectAssertions.cs

```diff
@@ -220,6 +220,15 @@ public AndConstraint<TAssertions> NotBe(TSubject unexpected, string because = ""
         return new AndConstraint<TAssertions>((TAssertions)this);
     }
 
+#pragma warning disable CA1065 // "Exceptions should not be raised in this type of method..."; this is intentional to ensure that FluentAssertions' users get a suggestion of what to use instead.
+    /// <inheritdoc/>
+    public override bool Equals(object obj) =>
+        throw new NotSupportedException("Equals is not part of Fluent Assertions. Did you mean Be() or BeSameAs() instead?");
+#pragma warning restore CA1065
+
+    /// <inheritdoc/>
+    public override int GetHashCode() => base.GetHashCode();
```

Comment:
The configured linting rules were upset that I overrode `Equals` without also overriding `GetHashCode()`.